### PR TITLE
[core] consolidate Dagster info for external resources in `DagsterRun.dagster_execution_info`

### PR DESCRIPTION
**Pull request by danielgafni**
## Summary & Motivation

There are a few Dagster components which create remote resources in order to execute code orchestrated from Dagster: 
- run launcher
- executor
- pipes invocations
- possibly, some other resources 

These components typically attach tags such as `dagster/run-id` or `dagster/code-location` to the launched remote resources. 

Until now there wasn't a single source of truth for these tags (however, I've recently added a helper property specifically for Pipes). 

I think it's a good idea to consolidate all these values in the `DagsterRun` since it's the common abstraction available in all three contexts. 

I'm also thinking we could add `InitExecutorContext.dagster_execution_info` and `OpExecutionContext.dagster_execution_info` which would have an additional `dagster/step-key` tag (not available on the `DagsterRun.dagster_execution_info`). 

## Changelog

[core] Dagster execution info (tags such as `dagster/run-id`, `dagster/code-location`, `dagster/user` and Dagster Cloud environment variables) typically attached to external resources is now available under `DagsterRun.dagster_execution_info`.

### CODE REVIEW

**Comment by danielgafni** [comment]
* **#26274** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26274?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26274?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by danielgafni** [comment]
@smackesey @gibsondan I encountered this issue while working on the ECS executor. Looks like all our run launchers and executors aren't really in sink on these Dagster tags. 

What do you think of this idea?

File: python_modules/dagster/dagster/_core/storage/dagster_run.py [hunk 1]
```diff
@@ -428,6 +430,56 @@ def get_root_run_id(self) -> Optional[str]:
     def get_parent_run_id(self) -> Optional[str]:
         return self.tags.get(PARENT_RUN_ID_TAG)
 
+    @cached_property
+    def well_known_tags(self) -> Dict[str, str]:
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/storage/dagster_run.py:434]: thinking through naming:
* i think the fact that `DagsterRun` has `tags` already means we should include something in the name that tries to clarify that these are not those / do not come from those 
* "well known" is a bit vague to me. From your PR summary "execution" / "orchestration" stand out. "system" and "context" come to mind for me too.

so like something in the direction of 
`derived_system_tags` 
`derived_execution_tags` 
`computed_origin_tags` 
`create_execution_tags` 
or avoid the word tags 
`system_info`
`origin_context`
`execution_labels`

**Review by alangenfeld — APPROVED** [review]
sounds good to me - is it under test via pipes already?

**Comment by danielgafni** [comment]
Yes, Pipes tests have checks for these tags being applied to external resources. 

Should we use this code in all run launchers and executors? This might cause small changes to tags here and there (e.g. `dagster/run_id` -> `dagster/run-id`) so we should probably do it through a deprecation cycle

**Pull request merged**